File: algorithm/java/backjoon/zero.java
package backjoon;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

class zero {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int num = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        for (int i=0; i < num; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int money = Integer.parseInt(st.nextToken());

            if (money == 0) {
                stack.pop();
            } else {
                stack.push(money);
            }
        }

        int answer = 0;
        for (int i : stack) {
            answer += i;
        }

        System.out.println(answer);
    }
}

File: algorithm/python/backjoon/stack/zero.py
import sys
In = sys.stdin.readline
N = int(In())
stack = []
add =0

for i in range(N):
    num = int(In())
    if num != 0:
        stack.append(num)
    else:
        stack.pop()

for i in stack:
    add += i
print(add)